src/CounterInterceptor/SimpleClientCallsInterceptor.cs

```diff
@@ -0,0 +1,97 @@
+namespace TemporalioSamples.CounterInterceptor;
+
+using Temporalio.Client;
+using Temporalio.Client.Interceptors;
+
+public record ClientCounts
+{
+    public uint Executions { get; internal set; }
+
+    public uint Signals { get; internal set; }
+
+    public uint Queries { get; internal set; }
+
+    public override string ToString() =>
+        $"\n\tTotal Number of Workflow Exec: {Executions}\n\t" +
+        $"Total Number of Signals: {Signals}\n\t" +
+        $"Total Number of Queries: {Queries}";
+}
+
+public class SimpleClientCallsInterceptor : IClientInterceptor
```

Comment:
IMO for simplicity you should make one interceptor that implements both `IClientInterceptor` and `IWorkerInterceptor` (and you only then need to configure the client not the worker interceptor separately). You could also make just one counts object.

Reply:
Done